ddspipe_core/src/cpp/communication/rpc/RpcBridge.cpp

```diff
@@ -310,6 +306,8 @@ void RpcBridge::transmit_(
                                                                     << ". Skipping data and continue.");
             continue;
         }
+    
+        RpcPayloadData& rpc_data = dynamic_cast<RpcPayloadData&>(*data);
```

Comment:
this is just a bug fix in mainline, i created another PR for this https://github.com/eProsima/DDS-Pipe/pull/142.